icicle/backend/cpu/src/curve/cpu_msm.hpp

```diff
@@ -780,12 +784,23 @@ eIcicleError cpu_msm_precompute_bases(
   const unsigned int shift = c * ((num_bms_no_precomp - 1) / precompute_factor + 1);
   for (int i = 0; i < nof_bases; i++) {
     output_bases[precompute_factor * i] = input_bases[i];
-    P point = P::from_affine(is_mont ? A::from_montgomery(input_bases[i]) : input_bases[i]);
+    // Handle required preprocess of base P according to the version of Field/Ec adder (accepting Barret / Montgomery)
+    P point =
+#ifdef BARRET
```

Comment:
maybe the degredation in performance comes from the Z=1 when moving from affine to projectve?
1 is a complex number and that might affect the multiplication by Z at the EC adder

Reply:
Degradation in the precompute?